controllers/util/onboarding_secrets.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ReadPrivateKey(cl client.Client) (*rsa.PrivateKey, error) {
```

Comment:
we aren't reading the key, we are getting it from API and parsing it as well, name `GetParsedPrivateKey` is more apt.

Reply:
ok